bytecomp/simplif.ml

```diff
@@ -235,6 +235,16 @@ let simplify_exits lam =
                                      ap_args=[x];
                                      ap_inlined=Default_inline;
                                      ap_specialised=Default_specialise}
+        (* Simplify %identity *)
+      | Pidentity, [e] -> e
```

Comment:
Why is this line part of the present change?

Reply:
The idiom being optimised is `Obj.with_tag (Obj.repr ( ... alloc ... ))` which has an extra `Pidentity` because `repr` expands to `%identity`. I thought that adding the general `Pidentity e -> e` transformation was cleaner than a composed `with_tag (Pidentity (Pmakeblock(...)) -> Pmakeblock(...)` transformation.